control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -97,6 +98,21 @@ func (r *NamespacedReconciler) ReconcileKind(ctx context.Context, broker *eventi
 
 	br := r.createReconcilerForBrokerInstance(broker)
 
+	statusConditionManager := base.StatusConditionManager{
+		Object:     broker,
+		SetAddress: broker.Status.SetAddress,
+		Env:        r.Env,
+		Recorder:   controller.GetEventRecorder(ctx),
+	}
+
+	// Get contract config map. Do this in advance, otherwise
+	// the dataplane pods that need volume mounts to the contract configmap
+	// will get stuck and will never be ready.
+	_, err := br.GetOrCreateDataPlaneConfigMap(ctx)
+	if err != nil {
+		return statusConditionManager.FailedToGetConfigMap(err)
+	}
```

Comment:
We don't need this since it will be created once we run `ReconcileKind`

Reply:
> // Get contract config map. Do this in advance, otherwise
   // the dataplane pods that need volume mounts to the contract configmap
   // will get stuck and will never be ready.

Or to put it differently, can you show me why this is true?

Reply:
If we don't get this configmap created before the dataplane is created by `br.ReconcileKind()`, then dataplane pods are stuck waiting for the configmap.

That's why I do this call in advance. It is idempotent anyway.

Reply:
They are not stuck because the CM is eventually created after a few miliseconds, so a warning event is not the same as being stuck

Reply:
No, they're actually stuck. I've seen them being stuck.

BTW, I updated my comment above https://github.com/knative-sandbox/eventing-kafka-broker/pull/2979#discussion_r1121395121 I was describing it wrong.

Reply:
That's not possible, it would work in CI currently on main

Reply:
What did you see when you say it was stuck? Can you describe how to reproduce the behavior?

Reply:
Hm, maybe they were stuck "only" because of a missing secret then... (which I solved later)

Let me check.

Reply:
@pierDipi You are right!

I removed this code now